Panic with go template

The following program `sample.go` triggers a panic:

```go
package main

import (
	"log"
	"os"
	"text/template"
)

type Message struct {
	Data string
}

func main() {
	tmpl := template.New("name")

	_, err := tmpl.Parse("{{.Data}}")
	if err != nil {
		log.Fatal(err)
	}

	err = tmpl.Execute(os.Stdout, Message{
		Data: "Hello, World!!",
	})

	if err != nil {
		log.Fatal(err)
	}
}
```

Expected result:
```console
$ go run ./sample.go
Hello, World!!
```

Got:
```console
$ yaegi ./sample.go
run.go:776: simple.go:21:32: genInterfaceWrapper error, no method Write
restricted.go:25: template: name:1:2: executing "name" at <.Data>: can't evaluate field Data in type stdlib._io_Writer
simple.go:14:10: panic
template: name:1:2: executing "name" at <.Data>: can't evaluate field Data in type stdlib._io_Writer
goroutine 1 [running]:
runtime/debug.Stack(0x1, 0xc0002e8600, 0x40)
	/Users/kevinpollet/.gvm/gos/go1.14.3/src/runtime/debug/stack.go:24 +0x9d
github.com/containous/yaegi/interp.(*Interpreter).Eval.func1(0xc00021bdd8)
	/Users/kevinpollet/Documents/Dev/yaegi/interp/interp.go:333 +0xbc
panic(0x191eae0, 0xc000204f30)
	/Users/kevinpollet/.gvm/gos/go1.14.3/src/runtime/panic.go:969 +0x166
github.com/containous/yaegi/interp.runCfg.func1(0xc0003660a0, 0xc000371a00)
	/Users/kevinpollet/Documents/Dev/yaegi/interp/run.go:119 +0x20c
panic(0x191eae0, 0xc000204f30)
	/Users/kevinpollet/.gvm/gos/go1.14.3/src/runtime/panic.go:969 +0x166
log.Panic(0xc000204ef0, 0x1, 0x1)
	/Users/kevinpollet/.gvm/gos/go1.14.3/src/log/log.go:351 +0xac
github.com/containous/yaegi/stdlib.logFatal(0xc000204ef0, 0x1, 0x1)
	/Users/kevinpollet/Documents/Dev/yaegi/stdlib/restricted.go:25 +0x3f
reflect.Value.call(0x191c2e0, 0x1a91c10, 0x13, 0x1a4e27d, 0x4, 0xc000209500, 0x1, 0x1, 0x18, 0xc000209500, ...)
	/Users/kevinpollet/.gvm/gos/go1.14.3/src/reflect/value.go:460 +0x8ab
reflect.Value.Call(0x191c2e0, 0x1a91c10, 0x13, 0xc000209500, 0x1, 0x1, 0x1a4b840, 0xc00020ce01, 0xc000209500)
	/Users/kevinpollet/.gvm/gos/go1.14.3/src/reflect/value.go:321 +0xb4
github.com/containous/yaegi/interp.callBin.func1(0x191c2e0, 0x1a91c10, 0x13, 0xc000209500, 0x1, 0x1, 0xc0002094e0, 0x1, 0x1)
	/Users/kevinpollet/Documents/Dev/yaegi/interp/run.go:1077 +0x64
github.com/containous/yaegi/interp.callBin.func10(0xc0003660a0, 0xc000222230)
	/Users/kevinpollet/Documents/Dev/yaegi/interp/run.go:1218 +0x1c2
github.com/containous/yaegi/interp.runCfg(0xc000371a00, 0xc0003660a0)
	/Users/kevinpollet/Documents/Dev/yaegi/interp/run.go:125 +0x6d
github.com/containous/yaegi/interp.(*Interpreter).run(0xc0002341a0, 0xc000371100, 0xc000366000)
	/Users/kevinpollet/Documents/Dev/yaegi/interp/run.go:103 +0x247
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc0002341a0, 0xc0002e02c0, 0x142, 0x0, 0x0, 0x0, 0x0, 0x0)
	/Users/kevinpollet/Documents/Dev/yaegi/interp/interp.go:416 +0x5f7
main.main()
	/Users/kevinpollet/Documents/Dev/yaegi/cmd/yaegi/yaegi.go:166 +0x6e2
```

--- a/_test/issue-772.go
+++ b/_test/issue-772.go
@@ -1,0 +1,31 @@
+package main
+
+import (
+	"log"
+	"os"
+	"text/template"
+)
+
+type Message struct {
+	Data string
+}
+
+func main() {
+	tmpl := template.New("name")
+
+	_, err := tmpl.Parse("{{.Data}}")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	err = tmpl.Execute(os.Stdout, Message{
+		Data: "Hello, World!!",
+	})
+
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+// Output:
+// Hello, World!!

--- a/interp/run.go
+++ b/interp/run.go
@@ -1080,7 +1080,7 @@ func callBin(n *node) {
 	}
 
 	for i, c := range child {
-		defType := funcType.In(pindex(i, variadic))
+		defType := funcType.In(rcvrOffset + pindex(i, variadic))
 		switch {
 		case isBinCall(c):
 			// Handle nested function calls: pass returned values as arguments

